units/sleep_scheduler.go

```diff
@@ -282,7 +299,21 @@ func (j *sleepSchedulerJob) makeStopAndStartJobs(ctx context.Context, _ evergree
 	hostIDsToStart := make([]string, 0, len(hostsToStart))
 	for i := range hostsToStart {
 		h := hostsToStart[i]
-		startJobs = append(startJobs, NewSpawnhostStartJob(&h, evergreen.ModifySpawnHostSleepSchedule, sleepScheduleUser, ts.Format(TSFormat)))
+		waitUntil := now.Add(time.Duration(i) * time.Second)
+		if waitUntil.After(latestWaitUntil) {
+			waitUntil = latestWaitUntil
```

Comment:
Perhaps we should consider extending `latestWaitUntil` to prevent rate limiting if we get for example >300 hosts that have the same schedule? Not sure if that's a likely scenario at all though so feel free to disregard if you think it's alright to do so.

Reply:
I think it wouldn't happen for a long while since there's currently 450 unexpirable hosts. We could adjust this if we notice that rate limiting comes back as an issue.